clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -2196,4 +2196,40 @@ def ZeroInitConstOp : CIR_Op<"llvmir.zeroinit", [Pure]>,
   let hasVerifier = 0;
 }
 
-#endif // MLIR_CIR_DIALECT_CIR_OPS
+def AsmATT : I32EnumAttrCase<"x86_att", 0>;
+def AsmIntel : I32EnumAttrCase<"x86_intel", 1>;
+
+def AsmFlavor : I32EnumAttr<
+  "AsmDialect",
+  "ATT or Intel",
+  [AsmATT, AsmIntel]> {
+  let cppNamespace = "::mlir::cir";
+}
+
+def CIR_InlineAsmOp : CIR_Op<"asm"> {
```

Comment:
Should this be marked `RecursiveMemoryEffects`? I feel like MLIR passes could eliminate these instructions easily without some indication that it contains side-effects.

Reply:
well, I'm not sure. But I marked as you said, no problem